skyportal/handlers/api/candidate.py

```diff
@@ -521,10 +583,8 @@ def grab_query_results_page(q, total_matches, page, n_items_per_page, items_name
         or (info["totalMatches"] == 0 and page != 1)
     ):
         raise ValueError("Page number out of range.")
-    info[items_name] = (
-        q.limit(n_items_per_page).offset((page - 1) * n_items_per_page).all()
-    )
-
+    s = slice((page - 1) * n_items_per_page, page * n_items_per_page)
```

Comment:
There could potentially be a very large number of candidates in the DB -- grabbing them all without a `limit` seems like a bad idea -- this could take a very long time

Reply:
Agreed. I tried for some time to do this on the database side, but I couldn't see a straightforward solution so I just settled for this for now. Just getting distinct Objs from the passed in query was putting the sorting out of order, at least the way I was trying to do the deduplication. I'll keep pushing on this.